pkg/controller/atlasproject/cloud_provider_access.go

```diff
@@ -0,0 +1,211 @@
+package atlasproject
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.uber.org/zap"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+func ensureProviderAccessStatus(ctx context.Context, customContext *workflow.Context, project *v1.AtlasProject, groupID string) workflow.Result {
+	roleStatuses := project.Status.DeepCopy().CloudProviderAccessRoles
+	roleSpecs := project.Spec.DeepCopy().CloudProviderAccessRoles
+
+	result, condition := syncProviderAccessStatus(ctx, customContext, roleSpecs, roleStatuses, groupID)
+	if result != workflow.OK() {
+		customContext.SetConditionFromResult(condition, result)
+		return result
+	}
+	customContext.SetConditionTrue(status.CloudProviderAccessReadyType)
```

Comment:
`UnsetCondition` should probably be somewhere close by :)

Reply:
Done. Thank you)